File: 프로그래머스/unrated/181839. 주사위 게임 1/주사위 게임 1.java
class Solution {
    public int solution(int a, int b) {
        int answer = 0;
        
        boolean x = a%2 == 1;
        boolean y = b%2 == 1;
        
        if (x && y) {
            answer = a*a + b*b;
        } else if (x || y) {
            answer = 2*(a+b);
        } else {
            answer = Math.abs(a-b);
        }   
        
        return answer;
    }
}

File: 프로그래머스/unrated/181839. 주사위 게임 1/주사위 게임 1.py
def solution(a, b):
    answer = 0
    
    x = a%2
    y = b%2
    
    if x and y: answer = a*a + b*b
    elif x or y: answer = 2*(a+b)
    else: answer = abs(a-b)
      
    return answer